cruise-control-metrics-reporter/src/main/java/com/linkedin/kafka/cruisecontrol/metricsreporter/CruiseControlMetricsUtils.java

```diff
@@ -130,4 +134,39 @@ public static void maybeUpdateConfig(Set<AlterConfigOp> configsToAlter,
       }
     }
   }
+
+  /**
+   * Retries the {@code Supplier<Boolean>} function while it returns {@code true} and for the specified max number of attempts.
+   * The delay between each attempt is computed as: delay = scaleMs * base ^ attempt
+   * @param function the code to call and retry if needed
+   * @param scaleMs the scale for computing the delay
+   * @param base the base for computing the delay
+   * @param maxAttempts the max number of attempts on calling the function
+   */
+  public static void retry(Supplier<Boolean> function, long scaleMs, int base, int maxAttempts) {
+    int attempts = 0;
+    boolean retry;
+    do {
+      retry = function.get();
+      if (retry) {
+        try {
+          if (++attempts == maxAttempts)
+            break;
+          Thread.sleep(scaleMs * (long) Math.pow(base, attempts));
```

Comment:
Can we save some CPU cycles by avoiding the use of `pow()` and instead update the `timeToSleep` at each round using a single multiplication?